lib/util/git/index.ts

```diff
@@ -361,23 +362,33 @@ export async function checkoutBranch(branchName: string): Promise<CommitSha> {
   }
 }
 
+async function fileList(
+  root: string,
+  exclude: string[] = [],
+  path = [],
+  results: string[] = []
+): Promise<string[]> {
+  const files = await ((fs.readdir(
+    [root.replace(/\/*$/g, ''), ...path].join('/'),
+    {
+      encoding: 'utf8',
+      withFileTypes: true,
+    } as never
+  ) as unknown) as Promise<Dirent[]>);
+  for (const file of files) {
+    if (file.isFile()) {
+      results.push([...path, file.name].join('/'));
+    } else if (file.isDirectory() && !exclude.includes(file.name)) {
+      await fileList(root, [], [...path, file.name], results);
+    }
+  }
+  return results;
+}
+
 export async function getFileList(): Promise<string[]> {
   await syncGit();
-  const branch = config.currentBranch;
   const submodules = await getSubmodules();
-  const files: string = await git.raw(['ls-tree', '-r', branch]);
-  // istanbul ignore if
-  if (!files) {
-    return [];
-  }
-  return files
-    .split('\n')
-    .filter(Boolean)
-    .filter((line) => line.startsWith('100'))
-    .map((line) => line.split(/\t/).pop())
-    .filter((file: string) =>
-      submodules.every((submodule: string) => !file.startsWith(submodule))
-    );
+  return fileList(config.localDir, ['.git', ...submodules]);
```

Comment:
Do we think this is enough, or is there the possibility that we missed something_